packages/plugin-compiler/src/main/kotlin/io/realm/compiler/RealmSchemaLoweringExtension.kt

```diff
@@ -73,13 +73,17 @@ class RealmSchemaLoweringExtension : IrGenerationExtension {
                 @Suppress("LongMethod")
                 override fun visitConstructorCall(expression: IrConstructorCall): IrExpression {
                     if (REALM_CONFIGURATION == expression.symbol.owner.returnType.classFqName &&
-                        expression.symbol.owner.isPrimary
+                        !expression.symbol.owner.isPrimary
                     ) {
-                        // substitute the public constructor call with the internal one, which will contain
-                        // also the populated companion map
-                        val internalConstructor =
+                        // There exist 3 constructors in RealmConfiguration
+                        // 1) Primary private constructor accepting all builder fields.
+                        // 2) Public secondary constructor accepting a schema as a Set.
+                        // 3) Internal secondary constructor accept the map of <KClass, RealmCompanion>.
+                        // In this step we are finding calls to constructor(2) and transform it into calls
+                        // to constructor(3).
+                        val internalConstructor: IrConstructorSymbol =
                             pluginContext.lookupConstructorInClass(REALM_CONFIGURATION) {
-                                !it.owner.isPrimary
+                                !it.owner.isPrimary && (it.owner.valueParameters[2].type.classFqName?.toString().equals("kotlin.collections.Map"))
```

Comment:
```suggestion
                                !it.owner.isPrimary
                                        && it.valueParameters.size == 3        
                                        && (it.owner.valueParameters[2].type.classFqName == KOTLIN_COLLECTIONS_MAP)
```
Don't forget to add `import io.realm.compiler.FqNames.KOTLIN_COLLECTIONS_MAP` at the begining